app/assets/javascripts/discourse/app/mixins/upload.js

```diff
@@ -39,8 +39,10 @@ export default Mixin.create({
 
   _initialize: on("didInsertElement", function() {
     const $upload = $(this.element);
-    const reset = () =>
+    const reset = () => {
       this.setProperties({ uploading: false, uploadProgress: 0 });
+      $upload.find(".hidden-upload-field").val("");
```

Comment:
do we need jquery for this ?

Reply:
We already use a lot of jQuery around here and rely on jQuery-File-Upload. It will not be going away anytime soon, but I agree with your comment so I will make the changes.